lib/dialects/mysql/query.js

```diff
@@ -92,7 +92,22 @@ class Query extends AbstractQuery {
       this.handleInsertQuery(data);
 
       if (!this.instance) {
-        result = data[this.getInsertIdField()];
+        // handle bulkCreate AI primiary key 
+        if (
+          data.constructor.name === 'ResultSetHeader'
+          && this.model
+          && this.model.autoIncrementField
+          && this.model.autoIncrementField === this.model.primaryKeyAttribute
+          && this.model.rawAttributes[this.model.primaryKeyAttribute]
+        ) {
+          const startId = data[this.getInsertIdField()];
+          result = [];
+          for (let i = startId; i < startId + data.affectedRows; i++) {
```

Comment:
There are some notes in the docs
http://dev.mysql.com/doc/refman/5.6/en/information-functions.html#function_last-insert-id

- LAST_INSERT_ID is per-connection
- It always returns the _first_ ID of a bulk insert, "to make it possible to reproduce easily the same INSERT statement against some other server"

Sorry that I overlooked this when reviewing, but I'm also on the fence whether this is a good idea after all. The answer you generally find on the internet is "not possible in MySQL".